File: C++/QueueArray.cpp
#include <iostream>
using namespace std;

int queue[10];
int front = -1;
int back = -1;

void enqueue(int val);
int dequeue();
int peek();
bool isEmpty();
void printList();

int main(){
	cout << "Hello There !" << endl;
	cout << isEmpty() << endl;
	
	enqueue(1);
	enqueue(3);
	enqueue(5);
	enqueue(7);
	enqueue(9);

	// cout << peek() << endl;
	// printList();
	cout << dequeue() << endl;
	//cout << peek() << endl;
	cout << isEmpty() << endl;
	cout << dequeue() << endl;
	cout << isEmpty() << endl;
	cout << dequeue() << endl;
	cout << dequeue() << endl;
	// cout << isEmpty() << endl;
	cout << dequeue() << endl;
	cout << isEmpty() << endl;
	return 0;
}

int dequeue(){
	if (front == -1 || back == -1){
		cout << "Queue Empty !" << endl;
	}
	int val = queue[front];
	front++;
	return val;
}

int peek(){
	if (front == -1){
		cout << "Queue Empty !" << endl;
	}
	return queue[front];
}

void enqueue(int val){
	if (back == 10){
		cout << "Queue Full !" << endl;
	}

	if (front == -1){
		front++;
		back++;
		queue[front] = val;	
	}
	else {
		back++;
		queue[back] = val;
	}
}

bool isEmpty(){
	if (front == -1){
		return true;
	}
	else if (front == back){
		false;
	}
	else {
		return false;
	}
}

void printList(){
	for(int i = 0; i < 10; i++){
		cout << queue[i] << endl;
	}
}

File: Go/QueueArray.go
package main
import "fmt"

var front int = -1
var back int = -1 
var queue = [10]int{}

func enqueue(int data){
	if (rear > len(queue)){
		fmt.Println("Queue Full")
	}

	if front == -1 && rear == -1 {
		front++
		rear++
		queue[rear] = data	
	} else {
		rear++
		queue[rear] = data
	}
}

func dequeue() int {
	if front == -1 && rear == -1 {
		fmt.Println("Queue Empty")
	}
	data := queue[front] 
	front++
	return data
}

func peek() int {
	if front == -1 && rear == -1 {
		fmt.Println("Queue Empty")
	}
	return queue[front]
}

func size() int {
	return len(queue)
}

func isEmpty() bool {
	if len(queue) == 0 {
		return true
	} else {
		return false
	}
}


func printList(){
	for i := 0; i < len(queue); i++ {
		fmt.Println(queue[i])
	}
}

func main(){
	fmt.Println(isEmpty())

	enqueue(1)
	enqueue(3)
	enqueue(5)
	enqueue(7)
	enqueue(9)

	printList()

	fmt.Println(isEmpty())
	fmt.Println(size())
	fmt.Println(peek())
	fmt.Println(dequeue())

	printList()
}
